app/models/client_opportunity_match.rb

```diff
@@ -453,6 +453,36 @@ def expire_all_notifications(on: 1.week.from_now.to_date)
     notifications.update_all(expires_at: on)
   end
 
+  def reset_and_destroy!
+    client.make_available_in(match_route: match_route)
+    opportunity.update! available_candidate: !opportunity.active_matches.exists?
+    opportunity.try(:voucher).try(:sub_program).try(:update_summary!)
+    expire_all_notifications
+    destroy
+    Matching::RunEngineJob.perform_later
+  end
+
+  def would_be_client_multiple_match
+    client_related_matches.on_route(match_route).active.exists? && match_route.should_prevent_multiple_matches_per_client
+  end
+
+  def would_be_opportunity_multiple_match
+    opportunity.active_matches.exists? && !match_route.allow_multiple_active_matches
+  end
+
+  def can_be_reopened?
+    return false if would_be_client_multiple_match
+    return false if would_be_opportunity_multiple_match
+
+    true
+  end
+
+  def reopen!
+    client.make_unavailable_in(match_route: match_route)
+    update(closed: false, active: true, closed_reason: nil)
+    current_decision.update(status: :pending)
```

Comment:
I *think* if the current decision was DND accepts match there's no mechanism to "backup."  So, this may need to move the match back 2 steps in some situations.

Reply:
I added the back button to this step.